components/camel-rabbitmq/src/test/java/org/apache/camel/component/rabbitmq/RabbitMQConsumerIntTest.java

```diff
@@ -121,15 +123,18 @@ public void sentMessageWithTimestampIsReceived() throws InterruptedException, IO
      */
     @Test
     public void sentMessageIsReceivedWithHeadersRouting() throws InterruptedException, IOException, TimeoutException {
-        //should only be one message that makes it through because only
-        //one has the correct header set
+        // Should only be one message that makes it through,
+        // because only one has the correct header set
         to.expectedMessageCount(1);
 
         Channel channel = connection().createChannel();
         channel.basicPublish(HEADERS_EXCHANGE, "", propertiesWithHeader("foo", "bar"), MSG.getBytes());
         channel.basicPublish(HEADERS_EXCHANGE, "", null, MSG.getBytes());
         channel.basicPublish(HEADERS_EXCHANGE, "", propertiesWithHeader("foo", "bra"), MSG.getBytes());
 
+        // Only one message should be received, waiting for some other messages
+        Thread.sleep(500);
```

Comment:
Maybe you could use Awaitility for this purpose.

Reply:
Thanks, fixed